### PR TITLE
Update index commands support

### PR DESCRIPTION
**Pull request by sunnyagain**
Update Index Assume Unchanged and No Assume Unchanged commands added.

### CODE REVIEW

**Comment by kemayo** [comment]
Could you update this to not change all the file modes? I don't like to merge dirty patches like that.

File: git/index.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        root = git_root(self.get_working_dir())
+        picked_file = self.results[picked]
+        if isinstance(picked_file, (list, tuple)):
+            picked_file = picked_file[0]
+        # first 1 character ia status code, the second is a space
+        picked_file = picked_file[2:]
+        self.run_command(['git', 'update-index', '--no-assume-unchanged', picked_file.strip('"')],self.rerun, working_dir=root)
+
+    def rerun(self, result):
+        self.run()
```
Comment (kemayo) [inline git/index.py]: Style guide (PEP8) complaint: file should end with a newline.

File: git/index.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            command += ['update-index', '--assume-unchanged']
+        command += ['--', picked_file]
+
+        self.run_command(command, self.rerun,
+            working_dir=working_dir)
+
+    def rerun(self, result):
+        self.run()
+
+class GitUpdateIndexNoAssumeUnchangedCommand(GitWindowCommand):
```
Comment (kemayo) [inline git/index.py]: Style guide (PEP8) complaint: two blank lines between classes.

File: git/index.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    def panel_done(self, picked):
+        if 0 > picked < len(self.results):
+            return
+        root = git_root(self.get_working_dir())
+        picked_file = self.results[picked]
+        if isinstance(picked_file, (list, tuple)):
+            picked_file = picked_file[0]
+        # first 1 character ia status code, the second is a space
+        picked_file = picked_file[2:]
+        self.run_command(['git', 'update-index', '--no-assume-unchanged', picked_file.strip('"')],self.rerun, working_dir=root)
```
Comment (kemayo) [inline git/index.py]: Missing a space after the comma at `,self.rerun`.

File: git/index.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        return len(item) > 0
+
+    def panel_done(self, picked):
+        if 0 > picked < len(self.results):
+            return
+        root = git_root(self.get_working_dir())
+        picked_file = self.results[picked]
+        if isinstance(picked_file, (list, tuple)):
+            picked_file = picked_file[0]
+        # first 1 character ia status code, the second is a space
```
Comment (kemayo) [inline git/index.py]: Typo: `ia` should probably be `is a`.

File: git/index.py (hunk 1 above)
Comment (sunnyagain) [inline git/index.py]: I am not python thats why missed it. corrected now.

**Comment by sunnyagain** [comment]
Updated for Style guide (PEP8) complaint and file mode by mistake changes.

File: git/index.py (hunk 1 above)
Comment (kemayo) [inline git/index.py]: That's fine; really, I should better document my expectations somewhere contributors are likely to come across them.

**Pull request merged**